_opensearch/point-in-time-api.md

```diff
@@ -0,0 +1,231 @@
+---
+layout: default
+title: Point in Time API
+nav_order: 58
+has_children: false
+parent: Point in Time
+---
+
+# Point in Time API
+
+Use the [Point in Time (PIT)]({{site.url}}{{site.baseurl}}/opensearch/point-in-time/) API to manage Points in Time. 
+
+---
+
+#### Table of contents
+- TOC
+{:toc}
+
+---
+
+## Create a PIT
+Introduced 2.4
+{: .label .label-purple }
+
+Creates a PIT. The `keep_alive` query parameter specifies how long to keep this PIT and is required.
+
+### Path and HTTP methods
+
+```json
+POST /<target_indexes>/_search/point_in_time?keep_alive=1h&routing=&expand_wildcards=&preference= 
+```
+
+### Path parameters
+
+Parameter | Data Type | Description 
+:--- | :--- | :---
+target_indexes | String | The name(s) of the target index(es) for the PIT. May contain a comma-separated list or a wildcard index pattern.
+
+### Query parameters
+
+Parameter | Data Type | Description
+:--- | :--- | :---
+keep_alive | Time |  The amount of time to keep the PIT. Required.
+preference | String | The node or the shard used to perform the search. Optional. Default is random.
+routing | String | Specifies to route search requests to a specific shard. Optional. Default is the document's `_id`. 
+expand_wildcards | String | The type of index that can match the wildcard pattern. Supports comma-separated values. Valid values are the following:<br>- `all`: Match any index or data stream, including hidden ones. <br>- `open`: Match open, non-hidden indexes or non-hidden data streams. <br>- `closed`: Match closed, non-hidden indexes or non-hidden data streams. <br>- `hidden`: Match hidden indexes or data streams. Must be combined with `open`, `closed` or both `open` and `closed`.<br>- `none`: No wildcard patterns are accepted.<br> Optional. Default is `open`.
+allow_partial_pit_creation | Boolean | Specifies whether to create a PIT with partial failures. Optional. Default is `false`.
+
+#### Sample request
+
+```json
+POST /my-index-1/_search/point_in_time?keep_alive=100m
+```
+
+#### Sample response
+
+```json
+{
+    "id": "o463QQEPbXktaW5kZXgtMDAwMDAxFnNOWU43ckt3U3IyaFVpbGE1UWEtMncAFjFyeXBsRGJmVFM2RTB6eVg1aVVqQncAAAAAAAAAAAIWcDVrM3ZIX0pRNS1XejE5YXRPRFhzUQEWc05ZTjdyS3dTcjJoVWlsYTVRYS0ydwAA",
+    "_shards": {
+        "total": 1,
+        "successful": 1,
+        "skipped": 0,
+        "failed": 0
+    },
+    "creation_time": 1658146050064
+}
+```
+
+### Response fields
+
+Field | Data Type | Description 
+:--- | :--- | :---  
+id | [Base64 encoded binary]({{site.url}}{{site.baseurl}}/opensearch/supported-field-types/binary) | The PIT ID.
+creation_time | long | The time the PIT was created in milliseconds since the epoch. 
+
+## List all PITs
```

Comment:
Can we also add a snippet where user can extend keep alive using search API.

```
```json
GET /_search
{
  "size": 10000,
  "query": {
    "match" : {
      "user.id" : "elkbee"
    }
  },
  "pit": {
    "id":  "46ToAwMDaWR5BXV1aWQyKwZub2RlXzMAAAAAAAAAACoBYwADaWR4BXV1aWQxAgZub2RlXzEAAAAAAAAAAAEBYQADaWR5BXV1aWQyKgZub2RlXzIAAAAAAAAAAAwBYgACBXV1aWQyAAAFdXVpZDEAAQltYXRjaF9hbGw_gAAAAA==", 
    "keep_alive": "100m" <--- //optional to extend a PIT's keep alive. The new expiry will be last accessed time + max ( request keep alive, current keep alive )
  },
  "sort": [ 
    {"@timestamp": {"order": "asc", "format": "strict_date_optional_time_nanos"}},
    {"_shard_doc": "desc"}
  ]
}
```
```